internal/quartzretry/quartzretry.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2021-2024 SUSE LLC
+
+// This package is needed because of https://github.com/reugn/go-quartz/issues/119
+// Once the issue is fixed we can just call scheduler operations without retries.
+package quartzretry
+
+import (
+	"time"
+
+	"github.com/reugn/go-quartz/quartz"
+)
+
+const maxSchedulerOperationRetries = 3
+
```

Comment:
I think the two exported functions should have Doc comments? https://go.dev/wiki/CodeReviewComments#doc-comments

Also, I wonder if instead of this bespoke mechanism, this could use https://pkg.go.dev/k8s.io/client-go/util/retry#OnError ?

Reply:
Changed to use `OnError` as suggested. 
I've kept the `quartzretry` package so it's easier to delete once the fix is released.

We'll need to delete the whole package and change:
```go
err = quartzretry.DeleteScheduledJobWithRetries(h.scheduler, gitRepoPollKey)
```
to
```go
err = h.scheduler.DeleteJob(gitRepoPollKey)
```

and 
```go
scheduledJob, err := quartzretry.GetScheduledJobWithRetries(h.scheduler, key)
```
to
```go
scheduledJob, err := h.scheduler.GetScheduledJob(key)
```